docs/python-api-intro.md

```diff
@@ -27,158 +27,152 @@ menu:
+| hive      | Support for the Hive metastore                                        |
+| pyarrow   | PyArrow as a FileIO implementation to interact with the object store  |
+| s3fs      | S3FS as a FileIO implementation to interact with the object store     |
+| zstandard | Support for zstandard Avro compresssion                               |
+| snappy    | Support for snappy Avro compresssion                                  |
 
-``` python
-filtered_scan = scan.filter("id=5")
-```
+## Catalog
 
-`Schema` projections can be applied against a `TableScan` by passing a list of column names.
+To instantiate a catalog:
 
 ``` python
-filtered_scan = scan.select(["col_1", "col_2", "col_3"])
-```
+>>> from pyiceberg.catalog.hive import HiveCatalog
+>>> catalog = HiveCatalog(name='prod', uri='thrift://localhost:9083/')
 
-Because some data types cannot be read using the python library, a convenience method for excluding columns from projection is provided.
-
-``` python
-filtered_scan = scan.select_except(["unsupported_col_1", "unsupported_col_2"])
-```
+>>> catalog.list_namespaces()
+[('default',), ('nyc',)]
 
+>>> catalog.list_tables('nyc')
+[('nyc', 'taxis')]
 
-Calls to configuration methods create a new `TableScan` so that each `TableScan` is immutable.
+>>> catalog.load_table(('nyc', 'taxis'))
+Table(identifier=('nyc', 'taxis'), ...)
+```
 
-When a scan is configured, `planFiles`, `planTasks`, and `Schema` are used to return files, tasks, and the read projection.
+And to create a table from a catalog:
 
 ``` python
-scan = table.new_scan() \
-    .filter("id=5") \
-    .select(["id", "data"])
-
-projection = scan.schema
-for task in scan.plan_tasks():
-    print(task)
+from pyiceberg.schema import Schema
+from pyiceberg.types import TimestampType, DoubleType, StringType, NestedField
+
+schema = Schema(
+    NestedField(field_id=1, name="datetime", field_type=TimestampType(), required=False),
+    NestedField(field_id=2, name="bid", field_type=DoubleType(), required=False),
+    NestedField(field_id=3, name="ask", field_type=DoubleType(), required=False),
+    NestedField(field_id=4, name="symbol", field_type=StringType(), required=False),
+)
+
+from pyiceberg.table.partitioning import PartitionSpec, PartitionField
+from pyiceberg.transforms import DayTransform
+
+partition_spec = PartitionSpec(
+    PartitionField(source_id=1, field_id=1000, transform=DayTransform(), name="datetime_day")
+)
+
+from pyiceberg.table.sorting import SortOrder, SortField
+from pyiceberg.transforms import IdentityTransform
+
+sort_order = SortOrder(
+    SortField(source_id=4, transform=IdentityTransform())
+)
+
+from pyiceberg.catalog.hive import HiveCatalog
+catalog = HiveCatalog(name='prod', uri='thrift://localhost:9083/')
+
+catalog.create_table(
+    identifier='default.bids',
+    location='/Users/fokkodriesprong/Desktop/docker-spark-iceberg/wh/bids/',
+    schema=schema,
+    partition_spec=partition_spec,
+    sort_order=sort_order
+)
+
+Table(
```

Comment:
I usually put result output in a separate pre box, but up to you.

Reply:
I don't have a strong opinion on it. I split this one out since it is quite big.